src/formatDate.js

```diff
@@ -8,7 +8,47 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  const dateParts = date.split(fromSeparator);
+  let year, month, day;
+
+  // Extract date parts according to fromFormat
+  fromFormat.forEach((format, index) => {
+    if (format === 'YYYY' || format === 'YY') {
+      year = dateParts[index];
+    } else if (format === 'MM') {
+      month = dateParts[index];
+    } else if (format === 'DD') {
+      day = dateParts[index];
+    }
+  });
+
+  // Convert year formats if necessary
+  if (fromFormat[0] === 'YYYY' && toFormat[0] === 'YY') {
+    year = year.slice(-2);
+  } else if (fromFormat[0] === 'YY' && toFormat[0] === 'YYYY') {
+    year = parseInt(year, 10) < 30 ? '20' + year : '19' + year;
```

Comment:
The use of 'parseInt' is redundant here since 'year' is already a string and you're concatenating it with another string. You can directly check if 'year < 30' without converting it to a number.